src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1+Builder.cs

```diff
@@ -246,15 +246,13 @@ public void AddRange(IEnumerable<T> items)
             {
                 Requires.NotNull(items, "items");
 
-                int count;
-                if (items.TryGetCount(out count))
+                int count = ImmutableExtensions.GetCount(ref items);
```

Comment:
If items isn't a list, isn't this going to allocate a list and copy it all to one?  That's very expensive.